.github/workflows/test.yml

```
@@ -0,0 +1,107 @@
+          ~/.cargo/git
+          target
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+    - name: Build project
+      run: make build
+
+    - name: Run unit tests
+      run: make test
+
+    - name: Setup Python
+      uses: actions/setup-python@v4
```

actionlint (1.7.8)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)